Update submodule hafs_graphics.fd/hrd_gplot

ghassan-alaka opened the pull request:
## Description of changes
This PR brings the latest upgrades, fixes, and other modifications from HRD's GPLOT graphics package into the HAFS-Community `develop` branch. The functionality has been extensively tested in the HAFS workflow by HRD employees (@ghassan-alaka @AndrewHazelton @lgramer) and will further stabilize the hrdgraphics task.

## Dependencies (optional)
The HAFS-Community fork of GPLOT was successfully updated and can be found here:
-[ hafs-community/GPLOT/pull/11 ](https://github.com/hafs-community/GPLOT/pull/11)

## Contributors (optional)
Extensive testing and upgrades were performed by @AndrewHazelton, @lgramer, and Mu-Chieh Ko at HRD. A big thanks to @BinLiu-NOAA who has guided us through the merge process.

## Tests conducted
Extensive testing on Jet, Hera, and Orion. This package will NOT run on WCOSS2 due to issues with deprecated and/or unavailable software on that machine.

## Application-level regression test status
N/A

ghassan-alaka committed:
Modified files to run the latest hrdgraphics based on HRD's GPLOT software.

ghassan-alaka committed:
Pull the latest GPLOT branch (feature/realtime2022) for the hrdgraphics task.

BinLiu-NOAA commented:
@ghassan-alaka, PR #171 is merged. Please sync this PR branch with the latest HAFS develop branch. Thanks! Meanwhile, there is no HAFS application/workflow level regression tests using hrd graphics yet. So, we will not need conduct regression tests for this PR.

BinLiu-NOAA committed:
Merge remote-tracking branch 'remotes/origin/develop' into feature/hrdgraphics_202212

ghassan-alaka committed:
Update .gitmodules

The purpose of this commit is to update the branch for submodule "GPLOT" to "support/HAFS". All updates were originally merged into "feature/realtime2022".

ghassan-alaka committed:
Updated the GPLOT submoule branch from 'feature/realtime2022' to 'support/HAFS'.

ghassan-alaka committed:
Merge branch 'feature/hrdgraphics_202212' of https://github.com/hafs-community/HAFS into feature/hrdgraphics_202212

ghassan-alaka commented:
@BinLiu-NOAA I think this PR is almost ready to be merged. Within the `hafs-community/gplot` repository, all changes were merged into the `support/HAFS` branch. Two key updates to this PR today:
1) updated _.gitmodules_ to point to the correct GPLOT submodule branch
2) committed the updated GPLOT files under _./sorc/hafs_graphics.fd/hrd_gplot_

I am requesting that @AndrewHazelton or @lgramer conduct a test of this updated `support/HAFS` branch within the HAFS workflow. It should behave exactly as the `feature/realtime2022` branch, but it would be good to be sure that nothing unexpected occurred during the merge.

BinLiu-NOAA commented on jobs/JHAFS_HRDGRAPHICS:
```diff
@@ -44,13 +44,14 @@ export CDATE=${CDATE:-${YMDH}}
 
 export SENDCOM=${SENDCOM:-YES}
-export SCRUBDATA=${SCRUBDATA:-YES}
+export SCRUBDATA=${SCRUBDATA:-NO}
```
@ghassan-alaka, could you revert this change? The default value of SCRUBDATA should be YES, unless absolutely wanted to otherwise be NO.

ghassan-alaka replied:
Agreed @BinLiu-NOAA . I reverted the change and commented on it below.

ghassan-alaka committed:
Update JHAFS_HRDGRAPHICS: set default value of SCRUBDATA to "YES".

ghassan-alaka reviewed (commented):
I reverted the default definition of SCRUBDATA to "YES" for the hrdgraphics task. Now this task is consistent with behavior of other HAFS workflow tasks.

BinLiu-NOAA reviewed (approved):
Looks good to me.
